website/blog/2019-07-22-presto-unlimited-mpp-database-at-scale.md

```diff
@@ -0,0 +1,123 @@
+---
+title: Presto Unlimited: Open-Source MPP Database at Scale
+author: Wenlei Xie
+authorURL: https://www.linkedin.com/in/wenleix/
+authorFBID: 681470066
+---
+
+Wenlei Xie, Andrii Rosa, Shixuan Fan, Rebecca Schlussel, Tim Meehan
+
+## Introduction 
+
+Presto is an open source distributed SQL query engine for running analytic queries against data sources of all sizes ranging from gigabytes to petabytes.
+
+Presto was originally designed for interactive use cases, however, after seeing the merit in having a single interface for both batch and interactive, it is now also used heavily for processing batch workloads. As a concrete example, more than 80% of new warehouse batch workloads at Facebook are developed on .Presto. Its flexible “connector” design makes it possible to run queries against heterogeneous data sources — such as joining together Hive and MySQL tables without preloading the data.
+
+However, memory-intensive (many TBs) and long-running (mulitple hours) queries have been major pain points for Presto users. It is difficult to reason how much memory queries will use and when it will hit memory limit, and failures in long-running queries cause landing time variance. To improve user experience and scale MPP Database to large ETL workloads, we started this Presto Unlimited project.
+
+## Grouped Execution
+
+[Grouped execution](https://github.com/prestodb/presto/pull/8951) was developed to scale Presto to run memory-intensive queries by leveraging table partitioning. 
+
+Consider the following query, where table `customer` and `orders` are already bucketed on `custkey`: 
+
+```
+SELECT ...
+FROM customer JOIN orders
+    USING custkey
+```
+
+Without grouped execution, the workers will build hash table using all the data on the build side (Table `orders`):
+
+![Ungrouped Execution](/img/blog/2019-07-22-presto-unlimited-mpp-database-at-scale/ungrouped.png)
```

Comment:
I'm trying to pre-view it, and the image doesn't show up. Most likely the syntax is incorrect.

Reply:
Actually nevermind. It's just Github doesn't understand the references.

Reply:
@arhimondr : Oh... I thought I opened the PR against my own repo .. But turns out it's against the master repo. Anyhow :D